src/plugins/intel_cpu/src/mkldnn_memory.cpp

```diff
@@ -115,10 +115,37 @@ void MKLDNNMemory::Create(MemoryDescPtr desc, const void* data, bool pads_zeroin
     }
 }
 
-void MKLDNNMemory::SetData(const MKLDNNMemory& src, size_t size, bool ftz) const {
-    MKLDNNReorderNode::reorderData(src, *this, size);
+void MKLDNNMemory::SetData(const MKLDNNMemory& src, bool ftz, bool dynBatch) const {
+    if (dynBatch) {
```

Comment:
Not sure I got why we need this special logic. Could you please elaborate?

Reply:
in legacy API when we work with dynamic batch, we passed input blobs with maximal batch in all infer request regardless real batch size which was set by `SetBatch`

but now we must passed input blobs with different batch size
and in case if we need to perform reordering we need to align blob desc with input node desc (which always has maximal batch size)
the same for output

Reply:
Ok, I see. The thing is that SetData is quite fundamental thing in the plugin. I would prefer to avoid introducing dirty WA inside core logic for not very important feature. From design perspective it is better to have special branches inside pushInput, pullOutput.

Reply:
extracted into pushInput, pullOutput.